theSrc/scripts/Quadrants.js

```diff
@@ -0,0 +1,56 @@
+import Plotly from 'plotly.js-basic-dist-min'
+
+async function drawQuadrants (plotly_chart, config) {
+    const layout = plotly_chart.layout
+    const x_range = plotly_chart._fullLayout.xaxis.range
+    const y_range = plotly_chart._fullLayout.yaxis.range
+    drawMidpointLines(layout, config, x_range, y_range)
+    await Plotly.relayout(plotly_chart, layout)
+}
+
+function drawMidpointLines (layout, config, x_range, y_range) {
+    // The range may be reversed, hence we still need to compute the min and max
+    const x_min = Math.min(...x_range)
+    const x_max = Math.max(...x_range)
+    const y_min = Math.min(...y_range)
+    const y_max = Math.max(...y_range)
+
+    if (config.xMidpoint >= x_min && config.xMidpoint <= x_max) {
```

Comment:
For consistency with the other axis lines, could you check for zero-width lines and hide the line if it is (the user may only want the shaded area to be visible)

Reply:
Thanks, done in https://github.com/Displayr/rhtmlCombinedScatter/pull/75/commits/3d3813abf51dbbee4b9e2429425b75be7953068f